tests/src/App/MappedElement.cpp

```diff
@@ -0,0 +1,129 @@
+{
+    // Arrange
+    Data::IndexedName indexedName {"EDGE1"};
+    Data::MappedName mappedName {"OTHER_NAME"};
+
+    // Act
+    Data::MappedElement mappedElement {indexedName, mappedName};
+
+    // Assert
+    EXPECT_EQ(mappedElement.index, indexedName);
+    EXPECT_EQ(mappedElement.name, mappedName);
+}
+
+TEST_F(MappedElementTest, moveConstructor)
+{
+    // Arrange
+    auto originalMappedElement = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto originalName = originalMappedElement.name;
+    auto originalIndex = originalMappedElement.index;
+
+    // Act
+    Data::MappedElement newMappedElement {std::move(originalMappedElement)};
+
+    // Assert
+    EXPECT_EQ(originalName, newMappedElement.name);
+    EXPECT_EQ(originalIndex, newMappedElement.index);
+}
+
+TEST_F(MappedElementTest, assignmentOperator)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE2", "ANOTHER_NAME");
+    EXPECT_NE(mappedElementA, mappedElementB);// Verify test setup
+
+    // Act
+    mappedElementA = mappedElementB;
+
+    // Assert
+    EXPECT_EQ(mappedElementA, mappedElementB);
+}
+
+TEST_F(MappedElementTest, moveAssignmentOperator)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE2", "ANOTHER_NAME");
+    EXPECT_NE(mappedElementA, mappedElementB);// Verify test setup
+
+    // Act
+    mappedElementA = std::move(mappedElementB);
+
+    // Assert
+    EXPECT_EQ(mappedElementA.name.toString(), "ANOTHER_NAME");
+    EXPECT_EQ(mappedElementA.index.toString(), "EDGE2");
+}
+
+TEST_F(MappedElementTest, equalityOperatorsWhenNotEqual)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE2", "ANOTHER_NAME");
+
+    // Act
+    bool aAndBAreEqual = mappedElementA == mappedElementB;
+    bool aAndBAreNotEqual = mappedElementA != mappedElementB;
+
+    // Assert
+    EXPECT_TRUE(aAndBAreNotEqual);
+    EXPECT_FALSE(aAndBAreEqual);
+}
+
+TEST_F(MappedElementTest, equalityOperatorsWhenEqual)
+{
+    // Arrange
+    auto mappedElementA = givenMappedElement("EDGE1", "OTHER_NAME");
+    auto mappedElementB = givenMappedElement("EDGE1", "OTHER_NAME");
+
+    // Act
+    bool aAndBAreEqual = mappedElementA == mappedElementB;
+    bool aAndBAreNotEqual = mappedElementA != mappedElementB;
+
+    // Assert
+    EXPECT_FALSE(aAndBAreNotEqual);
+    EXPECT_TRUE(aAndBAreEqual);
+}
+
+TEST_F(MappedElementTest, lessThanOperator)
+{
+    // Index is compared first, then mappedName
+
+    // Arrange
+    auto mappedElement1A = givenMappedElement("EDGE1", "A");
+    auto mappedElement1B = givenMappedElement("EDGE1", "B");
+    auto mappedElement2A = givenMappedElement("EDGE2", "A");
+    auto mappedElement2B = givenMappedElement("EDGE2", "B");
+    auto mappedElement2BDuplicate = givenMappedElement("EDGE2", "B");
```

Comment:
Some more questions which may be better in indexedname and mappedname but they are merged: what about, say "edge2" vs "edge12". Does it matter which is first in the order?

Reply:
They are sorted in numerical order (e.g. Edge2 comes before Edge12). I'm not sure if there is a test in IndexedName that verifies that, however.

Reply:
We may want to look in and add that. As such the test here is probably less important than there (the code is already as complex the test) IMO.

Reply:
https://github.com/FreeCAD/FreeCAD/pull/9048